File: Ejercicios/T4/CODIGO/C++/ContornosGradientes/laplaciana.cpp
#include <iostream>
#include <opencv2/imgproc.hpp>
#include <opencv2/highgui.hpp>

using namespace std;
using namespace cv;


int main(int argc, char** argv){

  //Lemos os argumentos de entrada
  const char* file = argv[1];
  Mat image = imread(file, IMREAD_GRAYSCALE);

  if(image.empty())
  {
    cout << "Erro lendo a imaxe" << file<< endl;
    return -1;
  }

Mat laplacian, LOG;
int kernelSize = 3;

// aplicamos a laplaciana
Laplacian(image, laplacian, CV_32F, kernelSize, 1, 0);

Mat img1;

GaussianBlur(image, img1, Size(3,3), 0, 0);

// Normalizamos as imaxes
normalize(laplacian, laplacian, 0, 1, NORM_MINMAX, CV_32F);

imshow("Laplaciana",laplacian);
waitKey(0);

return 0;
}

File: Ejercicios/T4/CODIGO/PYTHON/ContornosGradientes/laplaciana.py
import cv2
import numpy as np

img = cv2.imread("../../data/Contour.png", cv2.IMREAD_GRAYSCALE)
if img is None:
    print("Non poiden ler a imaxe")

kernelSize = 3

# Aplicamos a laplaciana
img1 = cv2.GaussianBlur(img,(3,3),0,0)
laplacian = cv2.Laplacian(img1, cv2.CV_32F, ksize = kernelSize,
                            scale = 1, delta = 0)

#Normalizamos os resultados
cv2.normalize(laplacian,
                dst = laplacian,
                alpha = 0,
                beta = 1,
                norm_type = cv2.NORM_MINMAX,
                dtype = cv2.CV_32F)

cv2.imshow("Laplaciana", laplacian)
cv2.waitKey(0)
